common/src/main/java/com/microsoft/identity/common/internal/ui/browser/BrowserSelector.java

```diff
@@ -29,60 +29,74 @@
 import android.content.pm.PackageInfo;
 import android.content.pm.PackageManager;
 import android.content.pm.ResolveInfo;
+import android.content.pm.Signature;
 import android.net.Uri;
 import android.os.Build;
+import android.util.Base64;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
-import com.microsoft.identity.common.java.exception.ClientException;
-import com.microsoft.identity.common.java.exception.ErrorStrings;
+import com.microsoft.identity.common.java.browser.Browser;
+import com.microsoft.identity.common.java.browser.IBrowserSelector;
 import com.microsoft.identity.common.java.ui.BrowserDescriptor;
 import com.microsoft.identity.common.java.util.StringUtil;
 import com.microsoft.identity.common.logging.Logger;
 import com.microsoft.identity.common.internal.broker.PackageHelper;
 
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
 
-public class BrowserSelector {
+public class BrowserSelector implements IBrowserSelector {
```

Comment:
Since this is in the Android common let's name it to AndroidBrowserSelector

```suggestion
public class AndroidBrowserSelector implements IBrowserSelector {
```